internal/datastore/postgres/readwrite.go

```diff
@@ -94,22 +98,26 @@ func (rwt *pgReadWriteTXN) WriteRelationships(mutations []*core.RelationTupleUpd
 	for _, mut := range mutations {
 		tpl := mut.Tuple
 
-		if tpl.Caveat != nil {
-			panic("caveats not currently supported in Postgres datastore")
-		}
-
 		if mut.Operation == core.RelationTupleUpdate_TOUCH || mut.Operation == core.RelationTupleUpdate_DELETE {
 			deleteClauses = append(deleteClauses, exactRelationshipClause(tpl))
 		}
 
 		if mut.Operation == core.RelationTupleUpdate_TOUCH || mut.Operation == core.RelationTupleUpdate_CREATE {
+			var caveatName string
+			var caveatContext map[string]any
+			if tpl.Caveat != nil {
+				caveatName = tpl.Caveat.CaveatName
+				caveatContext = tpl.Caveat.Context.AsMap()
+			}
 			valuesToWrite := []interface{}{
 				tpl.ResourceAndRelation.Namespace,
 				tpl.ResourceAndRelation.ObjectId,
 				tpl.ResourceAndRelation.Relation,
 				tpl.Subject.Namespace,
 				tpl.Subject.ObjectId,
 				tpl.Subject.Relation,
+				caveatName,
+				caveatContext,
```

Comment:
You're writing a `map` here? Shouldn't it be the bytes of the encoded context StructPb?

Please add a test for writing some context and reading it back... I think this is broken

Reply:
this is intentional. The PGX driver knows how to map a `map` into a `JSONB`. Remember, we are using JSONB so we can query the context later on without parsing it.

Reply:
Oh frick... can you add a comment here explaining that? it was totally non-obvious to me

Reply:
Sure can!